chemprop/v2/cli/train.py

```diff
@@ -32,41 +32,308 @@ class TrainSubcommand(Subcommand):
 
     @classmethod
     def add_args(cls, parser: ArgumentParser) -> ArgumentParser:
-        return add_args(parser)
+        parser = add_common_args(parser)
+        return add_train_args(parser)
 
     @classmethod
     def func(cls, args: Namespace):
         process_args(args)
+        validate_args(args)
         main(args)
 
+def add_common_args(parser: ArgumentParser) -> ArgumentParser:
+    parser.add_argument(
+        "--logdir",
+        nargs="?",
+        const="chemprop_logs",
+        help="runs will be logged to {logdir}/chemprop_{time}.log. If unspecified, will use 'save_dir'. If only the flag is given (i.e., '--logdir'), then will write to 'chemprop_logs'",
+    )
 
-def add_args(parser: ArgumentParser) -> ArgumentParser:
+    data_args = parser.add_argument_group("input data parsing args")
+    data_args.add_argument(
+        "-s",
+        "--smiles_columns",
+        type=list[str],
```

Comment:
```suggestion
        type=list,
```

Reply:
I assume this change is to allow users to specify the smiles columns by names (str) or by column index (int). I think that is a good idea.